src/state/opamSysInteract.ml

```diff
@@ -1093,14 +1105,50 @@ let install_packages_commands_t ?(env=OpamVariable.Map.empty) config sys_package
     [`AsUser (Commands.msys2 config),
      "-Su"::"--noconfirm"::packages], None
   | Netbsd -> [`AsAdmin "pkgin", yes ["-y"] ("install" :: packages)], None
+  | Nix ->
+    (match switch with
+    | None ->
+        log "Nix depext must be passed switch";
+        [], None
+    | Some switch ->
+    let dir = OpamPath.Switch.meta OpamStateConfig.(!r.root_dir) switch in
+    let drvFile = OpamFilename.create dir (OpamFilename.basename (OpamFilename.raw "env.nix")) in
+    let packages = String.concat " "
+      (OpamSysPkg.Set.fold (fun p l -> OpamSysPkg.to_string p :: l)
+      OpamSysPkg.Set.Op.(sys_packages ++ required) [])
+    in
+    let contents =
+{|{ pkgs ? import <nixpkgs> {} }:
+with pkgs;
+stdenv.mkDerivation {
+  name = "opam-nix-env";
+  nativeBuildInputs = with buildPackages; [ |} ^ packages ^ {| ];
```

Comment:
Is it possible to update that file for each installed package, like that there is no need to retrieve and pass the whole list of already installed packages and their system packages ?

Reply:
I think this is more ergonomic, as it allows system packages to be uninstalled too. It's bridging opam's imperative mode to nix's declarative model.